File: Data Structures & Algorithms/binary-tree-right-side-view/submission-0.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {TreeNode} root
     * @return {number[]}
     */
     res = [];
    rightSideView(root) {
        this.res=[];
        this.dfs(root, 0);
        return this.res;
    }
    dfs(node, depth) {
            if (!node) return;

            if (this.res.length === depth) {
                this.res.push(node.val);
            }

            this.dfs(node.right, depth + 1);
            this.dfs(node.left, depth + 1);
        }
}

File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    int result,k;
    public int kthSmallest(TreeNode root, int k) {
        this.k=k;
         dfs(root);
         return result;
    }
    public void dfs(TreeNode root){
        if(root==null)return;
        if (k==0){
            return;
        }
        dfs(root.left);
        k--;
        if(k==0){
        result=root.val;
        }
        dfs(root.right);
    }
}
